sklearn/datasets/openml.py

```diff
@@ -72,6 +90,20 @@ def _open_openml_url(openml_path, data_home):
     return gzip.GzipFile(local_path, 'rb')
 
 
+def _check_md5_checksum(fsrc, fdst, md5_checksum):
+
+    md5 = hashlib.md5()
+    block_size = 128 * md5.block_size
+    for block in iter(lambda: fsrc.read(block_size), b''):
+        md5.update(block)
+        fdst.write(block)
+
+    if md5_checksum != md5.hexdigest():
+        msg = 'Data set file hash {} does not match the checksum {}.'
+        msg = msg.format(md5.hexdigest(), md5_checksum)
+        raise Exception(msg)
```

Comment:
This does more than just *check* the MD5 hash of a file it also writes to a file stream. So it is at least named misleadingly.